src/main/java/thePackmaster/patches/GetEventNamePatch.java

```diff
@@ -0,0 +1,23 @@
+package thePackmaster.patches;
+
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePrefixPatch;
+import com.evacipated.cardcrawl.modthespire.lib.SpireReturn;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.helpers.EventHelper;
+import thePackmaster.SpireAnniversary5Mod;
+
+@SpirePatch(
+        clz = EventHelper.class,
+        method = "getEventName",
+        paramtypez = String.class
+)
+public class GetEventNamePatch {
```

Comment:
A patch like this is needed for every mod that wants custom events to show up properly in run history. Now that I think about it, some form of this should probably be in BaseMod, but it's an easy enough patch that nobody has bothered.